source/core/module_coreHelpers.js

```diff
@@ -56,15 +56,18 @@
             fpSync = fn;
         };
 
-        Module.coreHelpers.utf8ArrayToStringWithNull = function (u8Array, idxIn) {
+        // Takes Vireo Strings that are UTF-8 encoded strings with known length and returns a JS string
+        Module.coreHelpers.sizedUtf8ArrayToJSString = function (u8Array, startIndex, length) {
             /* eslint-disable no-continue, no-plusplus, no-bitwise */
             var u0, u1, u2, u3, u4, u5;
-            var idx = idxIn;
+            var idx = startIndex;
+            var lastVisitedIndex = startIndex + length;
```

Comment:
This is weird, implying length is a length of the remaining string starting at index, rather than the whole string, even though the whole string is passed in.  That seems like a bad API.